sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/implementation/CloudEventTracingPipelinePolicy.java

```diff
@@ -34,8 +35,17 @@ public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineN
         StringBuilder bodyStringBuilder = new StringBuilder();
         if (TracerProxy.isTracingEnabled() && contentType != null
             && Constants.CLOUD_EVENT_CONTENT_TYPE.equals(contentType.getValue())) {
-            return request.getBody().map(byteBuffer -> bodyStringBuilder.append(new String(byteBuffer.array(),
-                StandardCharsets.UTF_8)))
+            return request.getBody()
+                .map(byteBuffer -> {
+                    if (byteBuffer.hasArray()) {
+                        return bodyStringBuilder.append(new String(byteBuffer.array(),
+                            byteBuffer.arrayOffset() + byteBuffer.position(), byteBuffer.remaining(),
```

Comment:
Any reason for using this overload with offset and count over just `new String(byteBuffer.array())`?

Reply:
The backing array may not be fully written, ex

```java
byte[] bytes = new byte[1024];
bytes[0] = 42;

ByteBuffer buffer = ByteBuffer.wrap(bytes, 0, 1);
```

In this `ByteBuffer` only the first byte has been written but `.array()` will return the entire 1KB array.